### PR TITLE
Rename and modernize the udev plugin

### PR DESCRIPTION
**Pull request by hughsie**
The new plugin is called `optionrom` as this is the only type of image that it
parses for verification only. FuUdevDevice is also the generic parent already.

### CODE REVIEW

File: plugins/optionrom/README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
-UDev Support
-============
+OptionROM Support
+=================
 
 Introduction
 ------------
```
Comment (superm1) [inline plugins/optionrom/README.md]: Line 21 of this file probably needs updating too

Comment (hughsie) [reply plugins/optionrom/README.md]: It's still valid, no? It's just doing the additional instance ID in the device, not in the FuRom helper.

Comment (superm1) [reply plugins/optionrom/README.md]: I guess.

**Comment by hughsie** [comment]
>  I think we might need to have a post install script change for hand installed folks

I just did:

    fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_CONFLICTS, "udev");

and then the old plugin gets disabled. I'm somewhat hesitant to delete anything in a build script.

**Comment by superm1** [comment]
> and then the old plugin gets disabled. I'm somewhat hesitant to delete anything in a build script.

I was thinking a `rm -f` in `meson_post_install.sh`.  If it doesn't exist it won't be harmful.

**Comment by superm1** [comment]
> Well, the old plugin is going to have quirk files too, right? And possibly config files too. I'm not sure I care too much about deleting stuff in the users homedir. In the packaged case the old files gets removed on upgrade.

OK.

**Pull request merged**